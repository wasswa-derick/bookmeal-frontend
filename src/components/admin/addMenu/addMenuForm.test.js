import React from "react";
import { shallow } from "enzyme";
import AddMenuForm from "./addMenuForm";

describe("AddMenu", () => {
  let wrapper;
  beforeEach(() => {
    const menu = {
      id: 0,
      title: "",
      description: "",
      meals: []
    };
    const errors = {};
    const menuDate = {};
    wrapper = shallow(
      <AddMenuForm
        menu={menu}
        errors={errors}
        menuDate={menuDate}
        handleDateChange={jest.fn}
        handleFileUpload={jest.fn}
        onChange={jest.fn}
        onSubmit={jest.fn}
        checked={jest.fn}
        title="title"
      />
    );
  });

  it("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
