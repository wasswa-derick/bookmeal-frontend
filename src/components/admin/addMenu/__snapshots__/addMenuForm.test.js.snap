// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddMenu should render correctly 1`] = `
<div>
  <button
    className="btn-save btn btn-primary float-right"
    onClick={[Function]}
  >
    Save
  </button>
  <h4>
    title
  </h4>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-md-3"
    >
      <FormInput
        label="Title"
        name="title"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="col-md-3"
    >
      <div
        className="form-group"
      >
        <label
          htmlFor="date"
        >
          Menu Date
        </label>
        <DatePicker
          allowSameDay={false}
          className="form-control"
          dateFormat="L"
          dateFormatCalendar="MMMM YYYY"
          disabled={false}
          disabledKeyboardNavigation={false}
          dropdownMode="scroll"
          monthsShown={1}
          onBlur={[Function]}
          onChange={[Function]}
          onClickOutside={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMonthChange={[Function]}
          onSelect={[Function]}
          onYearChange={[Function]}
          preventOpenOnFocus={false}
          selected={Object {}}
          shouldCloseOnSelect={true}
          showTimeSelect={false}
          timeCaption="Time"
          timeIntervals={30}
          withPortal={false}
        />
      </div>
    </div>
    <div
      className="col-md-3"
    >
      <div
        className="form-group"
      >
        <label
          htmlFor="desc"
        >
          Description
        </label>
        <textarea
          className="form-control"
          cols="20"
          id="desc"
          name="description"
          onChange={[Function]}
          rows="2"
          value=""
        />
      </div>
    </div>
    <div
      className="col-md-3"
    >
      <div
        className="form-group"
      >
        <label
          htmlFor="menu-image"
        >
          Attach Image
        </label>
        <input
          accept="image/x-png,image/jpeg,image/png"
          className="form-control"
          name="users-file"
          onChange={[Function]}
          type="file"
        />
      </div>
    </div>
  </div>
  <div
    className="row"
  />
</div>
`;
